jax/api.py

```diff
@@ -1955,6 +1955,30 @@ def __init__(self, shape, dtype):
     self.shape = shape
     self.dtype = dtype
 
+  size = property(lambda self: onp.prod(self.shape))
+  ndim = property(lambda self: len(self.shape))
+
+  def __len__(self):
+    try:
+      return self.shape[0]
+    except IndexError:
+      raise TypeError("len() of unsized object")  # same as numpy error
+
+  def __repr__(self):
+    dtype = self.dtype.dtype
+    return f"{type(self).__name__}(shape={self.shape}, dtype={dtype.name})"
```

Comment:
We haven't yet dropped Python 3.5 compatibility. I don't think you can use f-strings yet.

Reply:
Fixed.